File: multi/1_Two_sum/Two_Sum.go
func twoSum(nums []int, target int) []int {
    var sums []int
    done := make(map[int]int)
    
    for i, _ := range nums {
        _, ok := done[target-nums[i]]
        if ok {
            sums = append(sums, done[target-nums[i]])
            sums = append(sums, i)
            return sums
        }
	    done[nums[i]] = i
    }
    return sums
}

File: multi/1_Two_sum/Two_Sum.java
import java.util.HashMap; 
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] sums = new int[]{0,1};
        Map<Integer, Integer> done = new HashMap<Integer, Integer>(); 
        int ex;

        for (int i = 0; i < nums.length; i++) {
            if(done.containsKey(target-nums[i])){
                sums = new int[]{done.get(target-nums[i]), i};
                break;
            }
            done.put(nums[i], i);
        }
        return sums;
    }
}